### PR TITLE
Move some static methods from HiveMetadata to HiveUtil

### PR DESCRIPTION
**Pull request by MiguelWeezardo**

### CODE REVIEW

**Comment by raunaqmorarka** [comment]
Should we move these to HiveUtil instead ?

**Comment by MiguelWeezardo** [comment]
> Should we move these to HiveUtil instead ?

Great idea.

**Review by findepi — APPROVED** [review]

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/AbstractTestHive.java [hunk 1]
```diff
@@ -2863,7 +2864,7 @@ public void testHideDeltaLakeTables()
                 .setOwner(session.getUser())
                 .setTableType(MANAGED_TABLE.name())
                 .setDataColumns(List.of(new Column("a_column", HIVE_STRING, Optional.empty())))
-                .setParameter(HiveMetadata.SPARK_TABLE_PROVIDER_KEY, HiveMetadata.DELTA_LAKE_PROVIDER);
+                .setParameter(HiveUtil.SPARK_TABLE_PROVIDER_KEY, HiveUtil.DELTA_LAKE_PROVIDER);
```
Comment (phd3) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/AbstractTestHive.java:2867]: nit: I'd change this to static import

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/AbstractTestHiveLocal.java [hunk 2]
```diff
@@ -223,7 +224,7 @@ private void markTableAsCreatedBySpark(SchemaTableName tableName, String provide
             ConnectorSession session = newSession();
             PrincipalPrivileges principalPrivileges = testingPrincipalPrivilege(session);
             Table oldTable = transaction.getMetastore().getTable(new HiveIdentity(session), tableName.getSchemaName(), tableName.getTableName()).get();
-            Table.Builder newTable = Table.builder(oldTable).setParameter(HiveMetadata.SPARK_TABLE_PROVIDER_KEY, provider);
+            Table.Builder newTable = Table.builder(oldTable).setParameter(HiveUtil.SPARK_TABLE_PROVIDER_KEY, provider);
```
Comment (phd3) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/AbstractTestHiveLocal.java:227]: nit: static import

**Review by phd3 — APPROVED** [review]
LGTM modulo a minor comment.

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/util/HiveUtil.java [hunk 3]
```diff
@@ -1049,4 +1052,25 @@ public static String sortingColumnToString(SortingColumn column)
     {
         return column.getColumnName() + ((column.getOrder() == DESCENDING) ? " DESC" : "");
     }
+
+    public static boolean filterSchema(String schemaName)
```
Comment (findepi) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/util/HiveUtil.java:1056]: i think i was writing a comment about this but cannot see it now... sorry for late thought then

this method is now public and no longer in context of use, we should give it a more descriptive name

`isHiveSystemSchema` (with inverse semantics)

**Comment by findepi** [comment]
@MiguelWeezardo see build outcome

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java [hunk 4]
```diff
@@ -243,11 +243,11 @@
 import static io.trino.plugin.hive.util.HiveBucketing.bucketedOnTimestamp;
 import static io.trino.plugin.hive.util.HiveBucketing.getHiveBucketHandle;
 import static io.trino.plugin.hive.util.HiveUtil.columnExtraInfo;
-import static io.trino.plugin.hive.util.HiveUtil.isHiveSystemSchema;
 import static io.trino.plugin.hive.util.HiveUtil.getPartitionKeyColumnHandles;
 import static io.trino.plugin.hive.util.HiveUtil.getRegularColumnHandles;
 import static io.trino.plugin.hive.util.HiveUtil.hiveColumnHandles;
 import static io.trino.plugin.hive.util.HiveUtil.isDeltaLakeTable;
+import static io.trino.plugin.hive.util.HiveUtil.isHiveSystemSchema;
```
Comment (findepi) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java:250]: Doesn't belong to `Move isIcebergTable to HiveUtils` commit

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/file/FileHiveMetastore.java [hunk 5]
```diff
@@ -557,7 +556,7 @@ public synchronized void renameTable(HiveIdentity identity, String databaseName,
         Table table = getRequiredTable(databaseName, tableName);
         getRequiredDatabase(newDatabaseName);
 
-        if (isIcebergTable(table.getParameters())) {
+        if (HiveUtil.isIcebergTable(table)) {
```
Comment (findepi) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/file/FileHiveMetastore.java:559]: static import

**Review by findepi — APPROVED** [review]

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java (hunk 4 above)
Comment (findepi) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java:250]: squash please :)

**Comment by MiguelWeezardo** [comment]
@findepi, @electrum could one of you merge this?

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java [hunk 6]
```diff
@@ -375,15 +375,15 @@ public SemiTransactionalHiveMetastore getMetastore()
     public List<String> listSchemaNames(ConnectorSession session)
     {
         return metastore.getAllDatabases().stream()
-                .filter(HiveMetadata::filterSchema)
+                .filter(schemaName -> !HiveUtil.isHiveSystemSchema(schemaName))
```
Comment (phd3) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java:378]: nit: `filter(not(HiveUtil::isHiveSystemSchema))`

**Review by phd3 — APPROVED** [review]
Can you please squash the fixup commit and also change the commit message to reflect that we're moving the methods instead of making them public? (Also, I'd avoid mentioning "redirection" in the commit message. We can just say that we're moving out static utility methods.) Will merge the PR once that's resolved.

(There's a minor comment, but would leave it up to you.)

**Comment by findepi** [comment]
squashed myself and merged

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java (hunk 6 above)
Comment (findepi) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java:378]: @phd3 sorry missed that, didn't refresh the page

**Pull request closed without merging**